### PR TITLE
Added header functionality

### PR DESCRIPTION
**Pull request by claudioLopezSA**

### CODE REVIEW

File: DesktopReact/frontend/src/components/FooterReplyModal/FooterReplyModal.js [hunk 1]
```diff
@@ -14,14 +14,15 @@ const style = {
     p: 4,
 };
 
+const body = "Duis mollis, est non commodo luctus, nisi erat porttitor ligula."
 const FooterReplyModal = () => {
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/FooterReplyModal/FooterReplyModal.js]: get from a fake http call

File: DesktopReact/frontend/src/components/post/Post.js [hunk 2]
```diff
@@ -7,11 +7,11 @@ import FooterReply from "../FooterReply/FooterReply";
 
 const Post = ({userName, postBody}) => {
 
-    const [collapse, setCollapse] = useState(true)
-    const Toggle = () => setCollapse(!collapse)
+    const [collapse, setCollapse] = useState(false)
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/post/Post.js:12]: should this always start un-collapsed?

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ import {
 import styled from "@emotion/styled";
 import FooterReply from "../FooterReply/FooterReply";
 import {actionsNav} from './interactionsConfig.json';
+import {toString} from "ip";
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: use JSON.stringify()

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 4]
```diff
@@ -24,89 +26,66 @@ const StyledBadge = styled(Badge)(({theme}) => ({
...
+const PostFooter = (props) => {
+    //console.log(props)
+    //const [isCollapse, setIsCollapse] = useState(false) // todo comment collapse function to listen the onClick
+    //const [isToggled, setIsToggled] = useState(); // TODO handle the reaction icons, if one pressed set it true and the others to false.
+    const [onHover, setOnHover] = useState(false)
+    const onHoverMouse = () => { // added onHover func to comment button, to hide it on reaction overmouse. Needed?
+        setOnHover(!onHover)
+    }
+
     const dialStyle = {
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js:33]: if styles don't in the css work use !important
ex. width: 100%!important;

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 5]
```diff
@@ -24,89 +26,66 @@ const StyledBadge = styled(Badge)(({theme}) => ({
...
+    //let badgeCounterValue = 1; // todo hardcoded values
+
+    // gets values from httpService.js array reactToPost function
+    const [badgeValues, setBadgeValues] = useState([])
+    const [likeValue, setLikeValue] = useState()
+
+    const BadgeCounterValue = () => {
+        let {data, result} = reactToPost();
+        //console.log(data);
+        setLikeValue(data[0][1])
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: if modifying the same value ex. likeValue++ use a callback
setLikeValue(value => value++) to modify it

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 6]
```diff
@@ -24,89 +26,66 @@ const StyledBadge = styled(Badge)(({theme}) => ({
...
+        let reactionsValue = data.filter((item) => item[0] !== 0).map(([i,k]) => [i,k]);
+        setBadgeValues(reactionsValue)
+
+    }
+    useEffect(() => {
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js:62]: unused code

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 7]
```diff
@@ -24,89 +26,66 @@ const StyledBadge = styled(Badge)(({theme}) => ({
...
     const HandleCommentContainer = () => {
-        setIsCollapse(isCollapse => !isCollapse)
-        // console.log("Hello from handleCommentContainer func: ")
-        // console.log(isCollapse)
-        console.log("Click comment button: ")
-        //FooterReply()
-
+        const {stateCallback} = props
+        stateCallback && stateCallback()
     }
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: why is this? it returns no value and calls the callback if exists?

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 8]
```diff
@@ -139,14 +119,15 @@ const PostFooter = () => {
...
-            tooltipTitle= {name}
+            tooltipTitle={name}
         />
     }
 
     const FooterComponent = () => {
         return <div className="footerCommentContainer">
             <Stack className="postFooterComment" direction="row">
                 <IconButton className="commentIconButton" size="small"
+                            //disabled={onHover}
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: comments inside a tag could break the code when compiled

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js [hunk 9]
```diff
@@ -166,23 +147,25 @@ const PostFooter = () => {
                     <SpeedDial
                         FabProps={{
                             style: {...dialStyle}
-                        }} // Access to the props of SpeedDial
+                        }} // Access to props of SpeedDial
                         color="secondary"
                         ariaLabel="SpeedDial"
+                        //onOpen={onHoverMouse}
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: comment inside the tag

File: DesktopReact/frontend/src/components/userProfileHeader/UserProfileHeader.js [hunk 10]
```diff
@@ -1,9 +1,33 @@
...
+import {Button, Card, CardContent, CardMedia, Typography} from "@mui/material";
 import banner from "../../images/banner.jpg";
 import pfp from "../../images/superalgos.png";
+import "./UserProfileHeader.css"
+import UserProfileModal from "./UserProfileModal";
+import {DateRangeOutlined, LocationOnOutlined} from "@mui/icons-material";
 
 const UserProfileHeader = ({actualUser}) => {
+
+    const profileIcons = { // this need to be moved to style css file
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/userProfileHeader/UserProfileHeader.js]: use !important in the css

File: DesktopReact/frontend/src/components/userProfileHeader/UserProfileHeader.js [hunk 11]
```diff
@@ -1,9 +1,33 @@
...
+        height: "15px",
+        verticalAlign: "text-top"
+    }
+
+    const buttonStyle = { // this need to be moved to style css file
+
+    }
+    const [modal, setModal] = useState(false);
+    const handleClick = () => setModal(!modal);
+        //console.log("hello from edit profile: ");
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/userProfileHeader/UserProfileHeader.js]: remove commented code thats useless, if will work soon leave it

File: DesktopReact/frontend/src/components/userProfileHeader/UserProfileModal.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
+import React, {useState} from 'react';
+import {Box, Button, CardContent, Modal, TextField, Typography} from "@mui/material";
+import "./UserProfileModal.css"
+import {CloseOutlined} from "@mui/icons-material";
+
+const style = { // todo need proper style
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/userProfileHeader/UserProfileModal.js]: css

File: DesktopReact/frontend/src/components/userProfileHeader/UserProfileModal.js [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+                                    </div>
+                                    <div className="editProfileHeaderTitleAndBtn">
+                                        <Typography className="editProfileTitle" variant="h5">
+                                        Edit Profile
+                                    </Typography>
+
+                                    </div>
+                                </div>
+                                <div className="editProfile">
+                                    <Typography className="styleASDASD" variant="subtitle2">
```
Comment (Loui-Molina) [inline DesktopReact/frontend/src/components/userProfileHeader/UserProfileModal.js]: nice name

**Review by Loui-Molina — CHANGES_REQUESTED** [review]
Little changes to do

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js (hunk 7 above)
Comment (claudioLopezSA) [reply DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: Nica's approach, validate what we pass. React trick

File: DesktopReact/frontend/src/components/post/Post.js (hunk 2 above)
Comment (claudioLopezSA) [reply DesktopReact/frontend/src/components/post/Post.js:12]: Flipped boolean value in the function, so is inverted in the state

File: DesktopReact/frontend/src/components/PostFooter/PostFooter.js (hunk 5 above)
Comment (claudioLopezSA) [reply DesktopReact/frontend/src/components/PostFooter/PostFooter.js]: Just needed the first value of the array. Still applies?

**Pull request merged**